istresearch/traptor
# PULSE-3508: Shuffle rules before performing a rule match

**BLUF**

Before track and follow Traptors perform a rule match, they will shuffle the redis rules (a list). Tests in a Jupyter Notebook show that a shuffle of up to 400 rules takes about 256 microseconds (performed with 1000 loops).

**Testing**

Get the code
- `git clone git@github.com:istresearch/traptor.git`
- `git checkout PULSE-3508`

Set up your environment
- Create and activate a virtual environment (venv or Anaconda)
- Rename traptor.env.sample to traptor.env and fill in all fields. Some defaults are provided.
- Install the requirements: `pip install -r requirements.txt`
- Fire up a local redis server (no longer a requirement with PULSE-3509)

Run the tests

- In the root directory: `python -m pytest -vv --cache-clear --cov=traptor --cov-report html`

All tests should pass.
